invenio_drafts_resources/services/draft.py

```diff
@@ -62,51 +56,52 @@ def search(self, querystring, identity, pagination=None, *args, **kwargs):
         # and simply configure DraftServiceConfig appropriately
         return self.resource_list_cls()
 
+    def _index_draft(self, draft):
+        indexer = self.indexer()
+        if indexer:
+            indexer.index(draft)
+
     def create(self, data, identity):
-        """Create a draft."""
-        # TODO: IMPLEMENT ME!
-        # # Check permissions: Can create draft of said record/resource
+        """Create a draft and the associated record (new)."""
         self.require_permission(identity, "create")
-        # Create record if new
-        # TODO: The `draft_of` class is set in the Draft API
-        # Creating the record class is business logic, but it is
-        # being set in the data access layer
-
-        # Get UUID, and version_id of forked record
-        # Create draft based on that
+        record = self.config.record_cls.create(data=data)
+        pid = self.minter()(record_uuid=record.id, data=record)  # Mint PID
+        marsh_data = self.data_validator().validate(data)
```

Comment:
nit: Syntactic sugar: isn't it nicer to have `minter`, `data_validator` as properties in the `RecordService` class so you can write e.g `self.data_validator.validate(data)`?

Reply:
Should we validate the `data` when we create a record? I remember that in the deposit workflow in the UI, we were having the case of creating an empty record when someone was uploading just a file. In that case, you have available only the file metadata but I assume that your validator it might require more fields I guess...Should we support partial validation when we create a draft/record? @lnielsen @fenekku

Reply:
Hmm I like the syntactic improvement. It would require refactor in invenio-records-resources tho (I can do it, maybe in a different PR - then adapted here)

As for validation. As it is now: It only validates according to the draft schema (I can re-order, to make the validation earlier). What we can do as "partial validation" is: Validate all existing fields (i.e. nothing - or almost nothing -  is required, but if they are they are validated). WDYT?

Reply:
I agree! I think when creating/saving a draft the partial behaviour would be to validate only what is passed in the incoming data through the marshmallow schema. Then, when you publish/edit you require full validation!

Reply:
For the syntactic improvement we can create an issue to address later if we like it :)

Reply:
Issues opened: 
- https://github.com/inveniosoftware/invenio-records-resources/issues/36
- https://github.com/inveniosoftware/invenio-drafts-resources/issues/6

> I agree! I think when creating/saving a draft the partial behaviour would be to validate only what is passed in the incoming data through the marshmallow schema. Then, when you publish/edit you require full validation!

Will refactor this after Im done gluing the service and views

Reply:
@zzacharo I have implemented the glue with the views in #7 I will tackle the data validation order in that PR to avoid this one hanging forever. I have set the `FIXME` there :)